data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/parser/ModelConvertingLogstashVisitor.java

```diff
@@ -96,11 +96,22 @@ else if (attributeContext.value().getChild(0) instanceof LogstashParser.PluginCo
         }
         else if (attributeContext.value().NUMBER() != null && attributeContext.value().getText().equals(attributeContext.value().NUMBER().toString())) {
             logstashValueType = LogstashValueType.NUMBER;
-            value = Double.valueOf(attributeContext.value().getText());
+            try {
+                value = Integer.parseInt(attributeContext.value().getText());
+            } catch (NumberFormatException e) {
+                try {
+                    value = Double.parseDouble(attributeContext.value().getText());
+                } catch (NumberFormatException exception) {
```

Comment:
I think we should throw `LogstashParsingException` here to indicate that .conf is not valid because number should either be an Integer or Double.

Reply:
Seems like a good idea.